### PR TITLE
fix(ui): fix references in base theme

### PR DESCRIPTION
**Pull request by dbanksdesign**
#### Description of changes

* Updated some theme references that were failing and added a test to make sure it won't happen in the future
* Updated style dictionary version
* Changing `defaultTheme` to be of type `BaseTheme` which makes more sense and is what people would use. 

#### Description of how you validated changes

tests pass and builds pass

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/ui/src/theme/index.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { createTheme } from './createTheme';
 import { baseTheme } from './baseTheme';
 
-export { createTheme };
+export { createTheme, baseTheme };
```
Comment (zchenwei) [inline packages/ui/src/theme/index.ts:4]: `baseTheme` has a confusing name `default-theme` 🤣

File: packages/ui/src/theme/tokens/components/copy.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 export const copy = {
-  fontSize: { value: '{font.sizes.xs.value}' },
-  gap: { value: '{space.relative.medium.value}' },
+  fontSize: { value: '{fontSizes.xs}' },
```
Comment (zchenwei) [inline packages/ui/src/theme/tokens/components/copy.ts:2]: Why `.value` could be skipped here? It seems we always need that when referencing other tokens value

Comment (dbanksdesign) [reply packages/ui/src/theme/tokens/components/copy.ts:2]: Correct, `.value` can be skipped now

**Review by jacoblogan — APPROVED** [review]
LGTM

File: packages/ui/src/theme/tokens/components/copy.ts (hunk 2 above)
Comment (reesscot) [reply packages/ui/src/theme/tokens/components/copy.ts:2]: +1 to this question, thought I was just missing something

Comment (reesscot) [reply packages/ui/src/theme/tokens/components/copy.ts:2]: Do we need to do this on all the other tokens too, or is it an optional change?

File: packages/ui/src/theme/__tests__/baseTheme.test.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { baseTheme } from '../baseTheme';
+import StyleDictionary from 'style-dictionary';
+
+describe('@aws-amplify/ui', () => {
+  describe('baseTheme', () => {
+    it('all references should resolve', () => {
+      expect(() => {
+        StyleDictionary.extend({
+          // This is to make typescript happy, the types don't jive well
+          tokens: Object.assign({ ...baseTheme.tokens }),
```
[the remark this answers is not in the document]
Comment (zchenwei) [reply packages/ui/src/theme/__tests__/baseTheme.test.ts:10]: Agree

File: packages/ui/src/theme/tokens/components/copy.ts (hunk 2 above)
Comment (dbanksdesign) [reply packages/ui/src/theme/tokens/components/copy.ts:2]: It is optional, we could do a PR to remove it for all the other tokens because it is cleaner

**Review by reesscot — APPROVED** [review]

**Pull request merged**